### DISCUSSION TITLE
Forward Vector for character

### DISCUSSION THREAD

**Discussion by ktimam** [jrouwe/JoltPhysics#868] [Q&A] [answered]
What is the best way to get the forward vector for Character?

**Comment by SirLynix** [comment, marked as answer]
Take its quaternion rotation and multiply it by your forward.
For example:
```cpp
JPH::Vec3 charForward = character.GetRotation() * JPH::Vec3(0, 0, -1); // assuming Y up and right-handed, (0,0,1) for left-handed
```

**Reply by eliasdaler** [reply]
(0, 0, 1) can still be "forward" in right-handed coordinate systems if your X axis points to the left - that’s the glTF coordinate system. :)

**Reply by jrouwe** [reply]
Note that there's a slightly more efficient version: `-character.GetRotation().RotateAxisZ()`
